src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -123,4 +121,39 @@ public void transform(StructuredRecord input,
                         Emitter<KeyValue<AvroKey<GenericRecord>, NullWritable>> emitter) throws IOException {
     emitter.emit(new KeyValue<>(new AvroKey<>(toAvroRecord(input)), NullWritable.get()));
   }
+
+  /**
+   * Validates configured output schema against underlying BigQuery table schema.
+   * @param context Sink Context
+   * @param collector Failure collector to report failures to the client.
+   */
+  private void validateConfiguredSchema(BatchSinkContext context, FailureCollector collector) {
```

Comment:
this is a good amount of duplicate code with prepareRunInternal.  It's kind of arbitrary that this class has a prepareRunInternal and prepareRunValidate. I think it's better to move the schema validation part to prepareRunInternal rather than try to keep this logic in the validate method.

This also would remove a duplicate schema lookup by passing the schema into overrideOutputSchemaWithTableSchemaIfNeeded instead of looking it up both here and in the method.

Reply:
I have moved code to prepareRunInternal and refactored code to avoid duplicate calls.